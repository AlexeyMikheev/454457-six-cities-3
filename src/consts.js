
export const Feature = {
  WIFI: `Wi-Fi`,
  WASHINGMACHINE: `Washing machine`,
  TOWELS: `Towels`,
  HEATING: `Heating`,
  COFFEEMACHINE: `Coffee machine`,
  BABYSET: `Baby seat`,
  KITCHEN: `Kitchen`,
  DISHWASHER: `Dishwasher`,
  CABELTV: `Cabel TV`,
  FRIDGE: `Fridge`
};

export const FEATURES = [
  Feature.WIFI,
  Feature.WASHINGMACHINE,
  Feature.HEATING,
  Feature.COFFEEMACHINE,
  Feature.BABYSET,
  Feature.DISHWASHER,
  Feature.CABELTV,
  Feature.FRIDGE
];

export const OfferType = {
  APARTMENT: `Apartment`,
  ROOM: `Private room`,
  HOUSE: `House`,
  HOTEL: `Hotel`
};

export const OFFERTYPES = [
  OfferType.APARTMENT,
  OfferType.ROOM,
  OfferType.HOUSE,
  OfferType.HOTEL
];

export const ViewMode = {
  Main: 0,
  Property: 1
};

export const VIEWMODES = [
  ViewMode.Main,
  ViewMode.Property
];

export const MIN_RATING = 1;
export const MAX_RATING = 5;

export const MAX_REVIEWS_DISPLAY_COUNT = 10;

export const MAX_IMAGES_DISPLAY_COUNT = 6;
export const MAX_NEAR_DISPLAY_COUNT = 3;

export const SortType = {
  POPULAR: `Popular`,
  PRICE_LH: `Price: low to high`,
  PRICE_HL: `Price: high to low`,
  TOPRATED: `Top rated first`
};

export const SORTTYPES = [
  SortType.POPULAR,
  SortType.PRICE_LH,
  SortType.PRICE_HL,
  SortType.TOPRATED
];

export const RatingValues = [5, 4, 3, 2, 1];

export const ErrorType = {
  UNAUTHORIZED: 401,
  BABREQUEST: 400
};

export const TIMEOUT = 5000;
export const URL = `https://htmlacademy-react-3.appspot.com/six-cities`;

export const AuthStatus = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`
};

export const LoadingStatus = {
  DEFAULT: `DEFAULT`,
  SUCCESS: `SUCCESS`,
  LOADING: `LOADING`,
  ERROR: `ERROR`
};

export const LoadingStatuses = [
  LoadingStatus.DEFAULT,
  LoadingStatus.SUCCESS,
  LoadingStatus.LOADING,
  LoadingStatus.ERROR
];

export const Url = {
  COMMENTS: `comments`,
  HOTELS: `hotels`,
  LOGIN: `login`,
  NEARBY: `nearby`
};

export const AppRoute = {
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER: `/offer`,
  ROOT: `/`,
};
