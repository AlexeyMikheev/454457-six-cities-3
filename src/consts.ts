export enum Feature {
    WIFI= `Wi-Fi`,
    WASHINGMACHINE= `Washing machine`,
    TOWELS= `Towels`,
    HEATING= `Heating`,
    COFFEEMACHINE= `Coffee machine`,
    BABYSET= `Baby seat`,
    KITCHEN= `Kitchen`,
    DISHWASHER= `Dishwasher`,
    CABELTV= `Cabel TV`,
    FRIDGE= `Fridge`
  }

export const FEATURES: Feature[] = [
  Feature.WIFI,
  Feature.WASHINGMACHINE,
  Feature.HEATING,
  Feature.COFFEEMACHINE,
  Feature.BABYSET,
  Feature.DISHWASHER,
  Feature.CABELTV,
  Feature.FRIDGE
];

export enum OfferType {
  APARTMENT= `Apartment`,
  ROOM= `Private room`,
  HOUSE= `House`,
  HOTEL= `Hotel`
}

export const OFFERTYPES: OfferType[] = [
  OfferType.APARTMENT,
  OfferType.ROOM,
  OfferType.HOUSE,
  OfferType.HOTEL
];

export enum ViewMode {
  Main= 0,
  Property= 1,
  Favorite= 2
}

// export const VIEWMODES: ViewMode[] = [
//   ViewMode.Main,
//   ViewMode.Property,
//   ViewMode.Favorite
// ];

export const MIN_RATING = 1;
export const MAX_RATING = 5;

export const MAX_REVIEWS_DISPLAY_COUNT = 10;

export const MAX_IMAGES_DISPLAY_COUNT = 6;
export const MAX_NEAR_DISPLAY_COUNT = 3;

export enum SortType {
  POPULAR= `Popular`,
  PRICE_LH= `Price: low to high`,
  PRICE_HL= `Price: high to low`,
  TOPRATED= `Top rated first`
}

export const SORTTYPES: SortType[] = [
  SortType.POPULAR,
  SortType.PRICE_LH,
  SortType.PRICE_HL,
  SortType.TOPRATED
];

export const RatingValues = [`5`, `4`, `3`, `2`, `1`];

export enum ErrorType {
  UNAUTHORIZED= 401,
  BABREQUEST= 400,
  NOTFOUND= 404
}

export const TIMEOUT = 5000;
export const URL = `https://htmlacademy-react-3.appspot.com/six-cities`;

export enum AuthStatus {
    NO_AUTH= `NO_AUTH`,
    LOADING= `LOADING`,
    AUTH= `AUTH`
  }

export enum LoadingStatus {
  DEFAULT= `DEFAULT`,
  SUCCESS= `SUCCESS`,
  LOADING= `LOADING`,
  ERROR= `ERROR`
}

export const LoadingStatuses: LoadingStatus[] = [
  LoadingStatus.DEFAULT,
  LoadingStatus.SUCCESS,
  LoadingStatus.LOADING,
  LoadingStatus.ERROR
];

export enum Url {
  COMMENTS= `comments`,
  HOTELS= `hotels`,
  FAVORITE= `favorite`,
  LOGIN= `login`,
  NEARBY= `nearby`
}

export enum AppRoute {
  LOGIN= `/login`,
  FAVORITES= `/favorites`,
  OFFER= `/offer`,
  ROOT= `/`,
}


export enum FavoriteState {
  MARKED= 1,
  UNMARKED= 0
}

export enum FavoriteButtonType {
  PROPERTY= 1,
  CARD= 0
}

// export const FavoriteButtonTypes = [
//   FavoriteButtonType.PROPERTY,
//   FavoriteButtonType.CARD
// ];

export enum ImageCardWidth {
  CARD_WIDTH= 260,
  CARD_HEIGHT= 200,
  FAVORITE_WIDTH= 150,
  FAVORITE_HEIGHT= 110
}

export const MIN_COMMENT_LENGTH = 50;
