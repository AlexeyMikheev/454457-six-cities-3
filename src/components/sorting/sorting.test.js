import React from "react";
import renderer from "react-test-renderer";
import Sorting from "./sorting.jsx";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {SortType} from "../../consts.js";

const mockStore = configureStore([]);

const mock = SortType.POPULAR;

it(`Render Sorting`, () => {

  const store = mockStore({
    offers: [],
    currentOffers: [],
    currentOffer: null,
    cities: [],
    reviews: [],
    currentCity: null,
    nearOffers: [],
    sortType: mock
  });

  const tree = renderer
    .create(<Provider store={store}><Sorting /></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
