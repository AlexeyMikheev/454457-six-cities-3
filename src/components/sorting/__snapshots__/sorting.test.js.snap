// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Sorting 1`] = `
Array [
  null,
]
`;
